app.js

```diff
@@ -1,180 +1,188 @@
-    if (!gitKey || !gitKey.keyPath) throw "Couldn't load ssh Key!";
-    didTurnAgentUp = await turnSshAgentUp(gitKey);
-  }
-  const addArgs = ["add"], commitArgs = [`commit -a -m "${commitMessage}"`];
-  if (overrideAdd.length > 0) addArgs.push(...overrideAdd);
-  else addArgs.push("-A");
-  const results = {};
-  try {
-    results.add = await execGitCommand(addArgs, path);
-    results.commit = await execGitCommand(commitArgs, path);
-    if (push) results.push = await execGitCommand(["push origin"], path);
-    return results;
-  }
-  catch (err) { throw { ...results, err }; }
-  finally {
-    if (didTurnAgentUp) await killSshAgent();
-  }
-}
-
-async function remove(action) {
-  const path = parsers.path(action.params.path);
-  await tryDelete(path);
-  return "Success";
-}
-
-module.exports = {
-  cloneUsingSsh,
-  clonePublic,
-  pull,
-  pushTag,
-  addCommit,
-  remove
-};
\ No newline at end of file
+const {
+  verifyGitVersion, execGitCommand, getSSHCommand, setUsernameAndEmail,
+  tryDelete, turnSshAgentUp, killSshAgent, isWin,
+} = require("./helpers");
+
+const parsers = require("./parsers");
+const GitKey = require("./git-key");
+
+async function cloneUsingSsh(action, settings) {
+  // verify git version
+  await verifyGitVersion();
+  // get parameters from action and settings
+  const path = parsers.path(action.params.path);
+  let repo = parsers.string(action.params.repo);
+  const branch = parsers.string(action.params.branch);
+  const {
+    overwrite, sshKey, extraArgs, saveCreds, username, password,
+  } = action.params;
+  const privateKey = sshKey || settings.sshKey;
+
+  // validate parameters
+  if (repo.startsWith("https://")) {
+    if (!username || !password) {
+      throw new Error("Both username and password are requried parameters for private repository URLs in HTTPS format.");
+    }
+    repo = `${repo.slice(0, 8)}${username}:${password}@${repo.slice(8)}`;
+  } else {
+    if (username || password) {
+      throw new Error("Parameters Username and Password are used only for repository URLs in HTTPS format.");
+    }
+    if (!privateKey) {
+      throw new Error("SSH key is required for repository URLs NOT in HTTPS format.");
+    }
+  }
+
+  // delete directory if already exists
+  if (overwrite) { await tryDelete(path); }
+  let gitKey = null;
+
+  const args = ["clone", repo];
+
+  if (branch) { args.push("-b", branch); }
+
+  if (privateKey && !(username && password)) { // if provided ssh Key and not username and password
+    const [newGitKey, sshCmd] = await getSSHCommand(privateKey, saveCreds);
+    gitKey = newGitKey;
+    args.push(`--config core.sshCommand="${sshCmd}"`);
+  }
+
+  if (extraArgs) { args.push(...parsers.array(extraArgs)); }
+  args.push(path);
+
+  // clone using key file
+  let didTurnAgentUp = false;
+  try {
+    if (gitKey) { didTurnAgentUp = await turnSshAgentUp(gitKey); }
+    // run clone
+    const cloneResult = await execGitCommand(args);
+    return cloneResult;
```

Comment:
We can directly return the result of the `execGitCommand`, no need to assign it to the constant. So we can replace it with `return execGitCommand(args)`.